chainntnfs/interface_test.go

```diff
@@ -548,6 +548,101 @@ func testBlockEpochNotification(miner *rpctest.Harness,
 	}
 }
 
+// testHistoricalBlockEpochNotification tests the case of multiple registered
+// client receiving historical block epoch  notifications due to their best known
+// block being out of date.
+func testHistoricalBlockEpochNotification(miner *rpctest.Harness,
+	notifier chainntnfs.TestChainNotifier, t *testing.T) {
+
+	const numBlocks = 10
+	const numClients = 5
+	var wg sync.WaitGroup
+	var extraBlockTest sync.WaitGroup
+
+	outdatedHash, outdatedHeight, err := miner.Node.GetBestBlock()
+	if err != nil {
+		t.Fatalf("unable to retrieve current height: %v", err)
+	}
+
+	// Generate numBlocks blocks which clients should receive historical
+	// notifications for.
+	if _, err := miner.Node.Generate(numBlocks); err != nil {
+		t.Fatalf("unable to generate blocks: %v", err)
+	}
+
+	// Ensure the neutrino notifier's best height is caught up to the chain.
+	// Otherwise, the clients will not receive all of the historical block
+	// notifications.
+	switch notifier := notifier.(type) {
+	case *neutrinonotify.NeutrinoNotifier:
+		targetHeight := uint32(outdatedHeight + numBlocks)
+		for totalMs := 0; notifier.GetBestHeight() != targetHeight &&
```

Comment:
No blocker, but a cleaner way to write this would be something like
```go
timeout := time.After(10 * time.Second)
for notifier.GetBestHeight() != targetHeight {
    select {
    case <-time.After(100 * time.Millisecond):
    case <-timeout:
        t.Fatalf(" error msg ")
    }
}
```

Reply:
n/a bc switched test to use `UnsafeStart`